### ISSUE TITLE
[BUG] Scrolling in the day Overview does not work

### ISSUE THREAD

**Issue by runkelruebe** [SecUSo/privacy-friendly-food-tracker#75]
If there are enough entries in the day Overview, the list gets cut off at the bottom of the screen and I can't scroll down, no matter where I try to scroll in the window (be it the middle or along the edges).

I **really** like the the app but this is a big problem especially when I need make changes to or want to delete an entry.

I am using a OnePlus 7 Pro with Oxygen OS 10.0.5.GM2 BA / Android 10, Security Patch Level March 1, 2020

**Comment by christianfl** [comment]
You could just read the last closed issue.

**Comment by runkelruebe** [comment]
Yup, my bad. I was a bit overzealous. Sorry for that!

Is there any info when the changes will be pushed to f-droid?

**Comment by christianfl** [comment]
That's a good question. Releases are not very regular here.

**Comment by runkelruebe** [comment]
That's sad. The fix is quite old now, three months if I saw correctly, one would assume it got published to the app stores. Is it a lack of time of the people who release it?

**Comment by runkelruebe** [comment]
Okay, so instead of giving up and after days of working my way around the missing scrolling, I tried to compile the app myself, because I **really** like the basic concepts of it and have already lost some weight.

When I start the app on my phone for the first time in debugging mode via Android studio, I get this exception
```
E/Perf: Fail to get file list org.secuso.privacyfriendlyfoodtracker
    getFolderSize() : Exception_1 = java.lang.NullPointerException: Attempt to get length of null array
E/Perf: Fail to get file list org.secuso.privacyfriendlyfoodtracker
E/Perf: getFolderSize() : Exception_1 = java.lang.NullPointerException: Attempt to get length of null array
```
but the tutorial is shown. So I swipe the tutorial and get to the "Preparation" screen. There, the loading wheel spins indefinitely and the three checkboxes stay empty.

I get this in the Debug console in Android studio:

`E/GenerateKeyActivity: Can't create handler inside thread Thread[Thread-3,5,main] that has not called Looper.prepare()`

If I click on the tick, I get to the Calendar view, regardless of the unsuccessful setup. When I tap on one of the days, the app basically crashes and goes back to the tutorial.
Part of the stack trace reads:

`W/System.err: java.lang.NullPointerException: Attempt to invoke virtual method 'byte[] java.lang.String.getBytes()' on a null object reference
        at android.util.Base64.decode(Base64.java:119)
        at org.secuso.privacyfriendlyfoodtracker.helpers.KeyGenHelper.getSecretKeyAsChar(KeyGenHelper.java:134)
W/System.err:     at org.secuso.privacyfriendlyfoodtracker.database.ApplicationDatabase.getInstance(ApplicationDatabase.java:58)
        at org.secuso.privacyfriendlyfoodtracker.ui.adapter.DatabaseFacade.<init>(DatabaseFacade.java:43)
        at org.secuso.privacyfriendlyfoodtracker.ui.OverviewActivity.getDbFacade(OverviewActivity.java:256)
        at org.secuso.privacyfriendlyfoodtracker.ui.OverviewActivity.refreshFoodList(OverviewActivity.java:535)`

So I cleared the storage for the app, disconnected my phone from Android Studio and started it again. The tutorial shows, I hit next, next and okay, and the three checkboxes for Generating Key, Generating Passphrase and Encrypting Database get set - yay, success! I hit the tick and get to the calendar. I tap on a day and get the Overview screen, hit the Plus on the bottom right, get the "Add a new entry" view. So far so good! Then I type in one character in the "Search for a product" input - aaaand the app crashes. It does this every time I try it. So after the successful setup, I hook my phone up to my computer and Adroid studio, start the app in debug mode from Android studio, select a day (some output here is `E/DatabaseFacade: Error o
E/DatabaseFacade: Error o` but no further specifics). I tap on the Plus, begin to type, and it crashes. The output is this:

`I/System.out: a
D/AndroidRuntime: Shutting down VM
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.secuso.privacyfriendlyfoodtracker, PID: 28666
    java.lang.IllegalStateException: Cannot access database on the main thread since it may potentially lock the UI for a long period of time.
        at android.arch.persistence.room.RoomDatabase.assertNotMainThread(RoomDatabase.java:204)
        at android.arch.persistence.room.RoomDatabase.query(RoomDatabase.java:232)
        at org.secuso.privacyfriendlyfoodtracker.database.ProductDao_Impl.findProductsByName(ProductDao_Impl.java:260)
        at org.secuso.privacyfriendlyfoodtracker.ui.adapter.DatabaseFacade.getProductByName(DatabaseFacade.java:198)
        at org.secuso.privacyfriendlyfoodtracker.ui.SearchFoodFragment$3.onTextChanged(SearchFoodFragment.java:232)
        at android.widget.TextView.sendOnTextChanged(TextView.java:10577)
        at android.widget.TextView.handleTextChanged(TextView.java:10667)
        at android.widget.TextView$ChangeWatcher.onTextChanged(TextView.java:13421)
        at android.text.SpannableStringBuilder.sendTextChanged(SpannableStringBuilder.java:1282)
        at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:591)
        at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:522)
        at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:42)
        at android.view.inputmethod.BaseInputConnection.replaceText(BaseInputConnection.java:843)
        at android.view.inputmethod.BaseInputConnection.setComposingText(BaseInputConnection.java:616)
        at com.android.internal.view.IInputConnectionWrapper.executeMessage(IInputConnectionWrapper.java:393)
        at com.android.internal.view.IInputConnectionWrapper$MyHandler.handleMessage(IInputConnectionWrapper.java:89)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7695)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
I/Process: Sending signal. PID: 28666 SIG: 9
Disconnected from the target VM, address: 'localhost:8601', transport: 'socket'
`

I can't really point to the changes from Version 1.0.1 that is in F-Droid and the current master that cause these problems.

Long story short:
It would be really f-ing amazing and appreciated if you guys could push a newer version to F-Droid.

**Comment by sleep-yearning** [comment]
Sorry for the silence from the developer team! 
We do have noticed and it is on our TODO list.

**Comment by runkelruebe** [comment]
I pulled the latest code, compiled, put it on my phone, and the bugs I reported are gone. Thanks!

**Comment by Kamuno** [comment]
Thanks for reporting. :)
